core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/ContractServiceExtension.java

```diff
@@ -118,18 +120,22 @@ private void registerServices(ServiceExtensionContext context) {
 
         var waitStrategy = context.hasService(NegotiationWaitStrategy.class) ? context.getService(NegotiationWaitStrategy.class) : new ExponentialWaitStrategy(DEFAULT_ITERATION_WAIT);
 
+        var telemetry = new Telemetry(GlobalOpenTelemetry.get());
```

Comment:
this doesn't seem to be the right place for this.

Reply:
Fixed. I have added the `telemetry` object to the Monitor interface. Have a look and let me know.

Reply:
Changed this to a more simple implementation. Now initializing the Telemetry object at the Runtime level.